presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -427,10 +428,11 @@ private static BingTile getTileCoveringLowerRightCorner(Envelope envelope, int z
         return tile;
     }
 
-    private static void checkGeometryToBingTilesLimits(OGCGeometry ogcGeometry, boolean pointOrRectangle, long tileCount)
+    private static void checkGeometryToBingTilesLimits(OGCGeometry ogcGeometry, boolean pointOrRectangle, long tileCount, int zoomLevel)
     {
         if (pointOrRectangle) {
-            checkCondition(tileCount <= 1_000_000, "The number of input tiles is too large (more than 1M) to compute a set of covering Bing tiles.");
+            checkCondition(tileCount <= 1_000_000, "%d input tiles is too large (more than 1M) to compute a set of " +
```

Comment:
perhaps, `The number of tiles covering input rectangle exceeds the limit of 1M. Number of tiles: %s. Rectangle: xMin=%s,  yMin=%s, xMax=%s, yMax=%s. Zoom level: %s.`

This way, we have a fixed prefix which allows for searching the code and details at the end of the message to provide some clues to the user.